### PR TITLE
Add a Recursive Check for HTML Entities

### PR DESCRIPTION
**Pull request by jplukarski**
Recursively checks for the presence of HTML Entities.

Co authored by @ibooker

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -24,12 +24,19 @@ export function sanitizeUrl(url?: string): string {
...
   }
-
-  const sanitizedUrl = decodeHtmlCharacters(url)
-    .replace(htmlCtrlEntityRegex, "")
-    .replace(ctrlCharactersRegex, "")
-    .trim();
-
+  let charsToDecode;
+  let decodedUrl = url;
+  do {
```
Comment (cgdibble) [inline src/index.ts:29]: Ohh fancy do/while! 🧐

**Review by cgdibble — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (jplukarski) [reply src/index.ts:29]: @ibooker 's idea!

**Review by SancSalix — APPROVED** [review]
⭐

**Comment by M4dHackers** [comment]
As this was an XSS bypass will a CVE be issued to notify consumers of the package?

Previous CVE's for bypasses: CVE-2022-48345, CVE-2021-23648

**Pull request merged**